cli/mender-artifact/modify_existing_test.go

```diff
@@ -287,45 +287,44 @@ func TestModifySigned(t *testing.T) {
 	err = ioutil.WriteFile(filepath.Join(tmp, "ecdsa.key"), []byte(PrivateECDSAKey), 0711)
 	assert.NoError(t, err)
 
-	// Create and sign artifact using RSA private key.
-	os.Args = []string{"mender-artifact", "write", "rootfs-image", "-t", "my-device",
-		"-n", "release-1", "-f", filepath.Join(tmp, "mender_test.img"),
-		"-o", filepath.Join(tmp, "artifact.mender"),
-		"-k", filepath.Join(tmp, "rsa.key")}
-	err = run()
-	assert.NoError(t, err)
+	for _, key := range []string{"rsa.key", "ecdsa.key"} {
 
-	// Modify the artifact and re-sign it using the same RSA private key.
-	os.Args = []string{"mender-artifact", "modify",
-		"-n", "release-2",
-		"-k", filepath.Join(tmp, "rsa.key"),
-		filepath.Join(tmp, "artifact.mender")}
+		// Create and sign artifact using RSA private key.
+		os.Args = []string{"mender-artifact", "write", "rootfs-image", "-t", "my-device",
+			"-n", "release-1", "-f", filepath.Join(tmp, "mender_test.img"),
+			"-o", filepath.Join(tmp, "artifact.mender"),
+			"-k", filepath.Join(tmp, key)}
+		err = run()
+		assert.NoError(t, err)
 
-	err = run()
-	assert.NoError(t, err)
+		// Modify the artifact, the result shall be unsigned
+		os.Args = []string{"mender-artifact", "modify",
+			"-n", "release-2",
+			filepath.Join(tmp, "artifact.mender")}
 
-	// Modify the artifact again, this time leaving it unsigned (no private key provided).
-	os.Args = []string{"mender-artifact", "modify",
-		"-n", "release-3",
+		err = run()
+		assert.NoError(t, err)
+
+		// Check for field update and unsigned state
+		os.Args = []string{"mender-artifact", "read",
 		filepath.Join(tmp, "artifact.mender")}
 
-	err = run()
-	assert.NoError(t, err)
+		r, w, err := os.Pipe()
+		out := os.Stdout
+		defer func() {
+			os.Stdout = out
+		}()
+		os.Stdout = w
 
-	// Create artifact again, this time with EC private key.
-	os.Args = []string{"mender-artifact", "write", "rootfs-image", "-t", "my-device",
-		"-n", "release-1", "-f", filepath.Join(tmp, "mender_test.img"),
-		"-o", filepath.Join(tmp, "artifact_ecdsa.mender"),
-		"-k", filepath.Join(tmp, "ecdsa.key")}
-	err = run()
-	assert.NoError(t, err)
+		go func() {
+			err = run()
+			assert.NoError(t, err)
+			w.Close()
+		}()
```

Comment:
Confused. You want to run this in the background? Doesn't it then return immediately, regardless of the result?

Reply:
I took this from other test that also tried to get the Stdout output.

I think that we run it in the background so that we can call ioutilReadAll(r), which will block until we call w.Close() from the goroutine.

Is there a simpler way to capture the Stdout here?

Reply:
No, you're right. I missed that part.